packages/components/src/Menu/Menu.tsx

```diff
@@ -106,80 +113,81 @@ export function Menu({ activator, items }: MenuProps) {
 
-  const menuClasses = classnames(
-    styles.menu,
-    position.vertical === "above" && styles.above,
-    position.vertical === "below" && styles.below,
-    position.horizontal === "left" && styles.left,
-    position.horizontal === "right" && styles.right,
-  );
+  // position related
+  const menuClasses = classnames(styles.menu);
 
   const wrapperClasses = classnames(styles.wrapper, {
     [styles.fullWidth]: fullWidth,
   });
 
   return (
-    <div
-      className={wrapperClasses}
-      ref={wrapperRef}
-      onClick={handleParentClick}
-    >
+    <div className={wrapperClasses} onClick={handleParentClick}>
+      <span ref={shadowRef} style={{ display: "none" }} />
       {React.cloneElement(activator, {
         onClick: toggle(activator.props.onClick),
         id: buttonID,
         ariaControls: menuID,
         ariaExpanded: visible,
         ariaHaspopup: true,
       })}
-      <AnimatePresence>
-        {visible && (
-          <>
-            <motion.div
-              className={styles.overlay}
-              onClick={toggle()}
-              variants={variation}
-              initial="overlayStartStop"
-              animate="done"
-              exit="overlayStartStop"
-              transition={{
-                type: "tween",
-                duration: 0.15,
-              }}
-            />
-            <motion.div
-              className={menuClasses}
-              role="menu"
-              aria-labelledby={buttonID}
-              id={menuID}
-              onClick={hide}
-              variants={variation}
-              initial="startOrStop"
-              animate="done"
-              exit="startOrStop"
-              custom={position}
-              transition={{
-                type: "tween",
-                duration: 0.25,
-              }}
-            >
-              {items.map((item, key: number) => (
-                <div key={key} className={styles.section}>
-                  {item.header && <SectionHeader text={item.header} />}
-
-                  {item.actions.map((action, index) => (
-                    <Action
-                      sectionLabel={item.header}
-                      key={action.label}
-                      shouldFocus={key === 0 && index === 0}
-                      {...action}
-                    />
+      <MenuPortal>
+        <AnimatePresence>
+          {visible && (
+            <>
+              <motion.div
+                className={styles.overlay}
+                onClick={toggle()}
+                variants={variation}
+                initial="overlayStartStop"
+                animate="done"
+                exit="overlayStartStop"
+                transition={{
+                  type: "tween",
+                  duration: 0.15,
+                }}
+              />
+              <div
+                ref={setPopperElement}
+                className={styles.popperContainer}
+                {...positionAttributes}
+                {...focusAttribute}
```

Comment:
this turns into `data-atl-maintain-portal-focus: "true"` on this wrapping div

I'm not super in love with this approach but it's the lightestweight solution I can think of that lends itself to expanding to other components that run into this same problem

Reply:
also I'm always putting this in here. might it be worth having more control over?

does get kind of weird exposing a prop like `shouldMenusMenuKeepFocusWhenYouLaunchItFromInputTextsToolbar` that is deeply connected to another component's implementation. I suppose we're doing that either way, but doing it secretly is better?